File: support_tools/swigwin-3.0.12/Examples/test-suite/python/varargs_runme.py
import varargs

if varargs.test("Hello") != "Hello":
    raise RuntimeError, "Failed"

f = varargs.Foo("Greetings")
if f.str != "Greetings":
    raise RuntimeError, "Failed"

if f.test("Hello") != "Hello":
    raise RuntimeError, "Failed"


if varargs.test_def("Hello", 1) != "Hello":
    raise RuntimeError, "Failed"

if varargs.test_def("Hello") != "Hello":
    raise RuntimeError, "Failed"

###
if varargs.test_plenty("Hello") != "Hello":
    raise RuntimeError, "Failed"

if varargs.test_plenty("Hello", 1) != "Hello":
    raise RuntimeError, "Failed"

if varargs.test_plenty("Hello", 1, 2) != "Hello":
    raise RuntimeError, "Failed"

try:
    varargs.test_plenty("Hello", 1, 2, 3)
    raise RuntimeError
except NotImplementedError:
    pass
except TypeError:
    pass

File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/varargs_runme.js
var varargs = require("varargs");

if (varargs.test("Hello") != "Hello") {
  throw new Error("Failed");
}

var f = new varargs.Foo("Greetings")
if (f.str != "Greetings") {
  throw new Error("Failed");
}

if (f.test("Hello") != "Hello") {
  throw new Error("Failed");
}

if (varargs.test_def("Hello",1) != "Hello") {
  throw new Error("Failed");
}

if (varargs.test_def("Hello") != "Hello") {
  throw new Error("Failed");
}

if (varargs.test_plenty("Hello") != "Hello") {
  throw new Error("Failed");
}

if (varargs.test_plenty("Hello", 1) != "Hello") {
  throw new Error("Failed");
}

if (varargs.test_plenty("Hello", 1, 2) != "Hello") {
  throw new Error("Failed");
}

var thrown = false;
try {
  varargs.test_plenty("Hello", 1, 2, 3);
} catch (err) {
  thrown = true;
}
if (!thrown) {
  throw new Error("Failed");
}
